library/core/src/any.rs

```diff
@@ -610,6 +610,84 @@ impl dyn Any + Send + Sync {
 /// While `TypeId` implements `Hash`, `PartialOrd`, and `Ord`, it is worth
 /// noting that the hashes and ordering will vary between Rust releases. Beware
 /// of relying on them inside of your code!
+///
+/// **Note on Variance**:
+/// Suppose `SubType` is a subtype of `SuperType`,
+/// that is, `SubType` can be used wherever `SuperType` can be used,
+/// and can be freely as-casted to it,
+/// and `CoVar`is a generic data type that has a covariant type parameter.
+/// Then by covariance, `CoVar<SubType>` is a subtype of `CoVar<SuperType>`,
+/// that is, `CoVar<SubType>` can be used wherever `CoVar<SuperType>` can be used,
+/// and can be freely as-casted to it.
+/// Then if `CoVar<SuperType>` relies on `TypeId::of::<SuperType>` to uphold any invariants,
```

Comment:
```suggestion
/// Then if `CoVar<SuperType>` relies on `TypeId::of::<SuperType>()` to uphold any invariants,
```

Reply:
I even made that mistake in my code once...